### PR TITLE
refactor TestGetRecentUnmetScheduleTimes in util_test.go

### PR DESCRIPTION
**Pull request by wccsama**
**What type of PR is this?**
```
/kind cleanup
```
**What this PR does / why we need it**:
```
refactor TestGetRecentUnmetScheduleTimes
```
**Which issue(s) this PR fixes**:

```
NONE
```
**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Comment by wccsama** [comment]
/assign soltysh
PTAL, thx~

**Comment by wccsama** [comment]
/assign @soltysh 
PTAL, thx~

File: pkg/controller/cronjob/utils_test.go [hunk 1]
```diff
@@ -268,114 +268,126 @@ func TestGetRecentUnmetScheduleTimes(t *testing.T) {
...
-		if err != nil {
-			t.Errorf("unexpected error: %v", err)
-		}
-		if len(times) != 1 {
-			t.Errorf("expected 1 start times, got: %v", times)
-		} else if !times[0].Equal(T2) {
-			t.Errorf("expected: %v, got: %v", T1, times[0])
+
+	for _, testCase := range testCases {
+		sj.ObjectMeta.CreationTimestamp = testCase.creationTimestamp
```
Comment (soltysh) [inline pkg/controller/cronjob/utils_test.go:361]: While you're at it, why not using `t.Run` here, which allows even greater flexibility around which sub-tests to run.

File: pkg/controller/cronjob/utils_test.go [hunk 2]
```diff
@@ -268,114 +268,126 @@ func TestGetRecentUnmetScheduleTimes(t *testing.T) {
...
+		sj.Status.LastScheduleTime = testCase.lastScheduleTime
+		sj.Spec.StartingDeadlineSeconds = testCase.deadline
+
+		times, err := getRecentUnmetScheduleTimes(sj, testCase.nowTime)
+		if err != nil && !testCase.expectErr {
+			t.Errorf("unexpect getRecentUnmetScheduleTimes err: %v, caseName: %v", err, testCase.caseName)
```
Comment (soltysh) [inline pkg/controller/cronjob/utils_test.go:367]: When using `t.Run` you won't have to deal with `caseName` in the logs.

File: pkg/controller/cronjob/utils_test.go [hunk 3]
```diff
@@ -268,114 +268,126 @@ func TestGetRecentUnmetScheduleTimes(t *testing.T) {
...
-			t.Errorf("expected 2 start times, got: %v", times)
-		} else {
-			if !times[0].Equal(T1) {
-				t.Errorf("expected: %v, got: %v", T1, times[0])
-			}
-			if !times[1].Equal(T2) {
-				t.Errorf("expected: %v, got: %v", T2, times[1])
-			}
+
+		if !equal(times, testCase.expectResult) {
```
Comment (soltysh) [inline pkg/controller/cronjob/utils_test.go:374]: Use `reflect.DeepEqual` instead`

**Review by soltysh — CHANGES_REQUESTED** [review]

File: pkg/controller/cronjob/utils_test.go (hunk 2 above)
Comment (wccsama) [reply pkg/controller/cronjob/utils_test.go]: Already fixed.

File: pkg/controller/cronjob/utils_test.go (hunk 3 above)
Comment (wccsama) [reply pkg/controller/cronjob/utils_test.go]: Already fixed.

File: pkg/controller/cronjob/utils_test.go (hunk 2 above)
Comment (wccsama) [reply pkg/controller/cronjob/utils_test.go]: Already fixed.

File: pkg/controller/cronjob/utils_test.go (hunk 1 above)
Comment (wccsama) [reply pkg/controller/cronjob/utils_test.go]: Already fixed.

**Comment by wccsama** [comment]
@soltysh 
Already used t.Run and reflect.DeepEqual, please take a look, thanks.

**Pull request closed without merging**